### ISSUE TITLE
Document minimum (for evaluation) and optimal configurations for example workloads

### ISSUE THREAD

**Issue by dcaddick** [Azure/Moodle#47]
Hi,

Just reporting that this template needs a min of 9 Cores.
So as it stands it will not run successfully from a Free Trial account unless you set:
- fileServerType:  nfs (instead of gluster)
- autoScaleVmSku: Standard_DS1_v2 (instead of DS2)
This will reduce the core requirements to under the max of 4 for a Free Trial account

> This repo contains guides on how to deploy and manage a scalable Moodle cluster on Azure. You can use an Azure free account to get started.

**Comment by rgardler** [comment]
You are indeed correct @dcaddick. The good news is that, as you note, it is now possible to configure many aspects of the template to reduce the minimum resources required to deploy a cluster. Unfortunately, as is so often the case the documentation is trailing what is possible in the template.

I'm going to change the subject of this issue to remind us to fix that. Pull requests are welcome...

We need to add documentation to:

- [ ] identify the minimum possible deployment configuration
- [ ] provide guidance on choosing an optimal configuration based on expected load
- [ ] document all configuration options in the template

**Comment by rgardler** [comment]
See an example config at https://github.com/Azure/Moodle/compare/condorcet#diff-8303e47b1360eff546d42ed10d135047

**Comment by hosungsmsft** [comment]
This is partially addressed in the [tshirt-sizing branch](https://github.com/Azure/Moodle/tree/tshirt-sizing). More refinement/improvement should be done before releasing it to master.

**Comment by rgardler** [comment]
t-shirt sizing branch now merged into master
